dove/tests/test_cmd_helper.rs

```diff
@@ -0,0 +1,162 @@
+pub fn project_new_with_args(
+    base_folder: &Path,
+    project_folder: &Path,
+    project_name: &str,
+    project_dialect: &str,
+    project_address: &str,
+    project_blockchain_api: &str,
+) {
+    let args = &[
+        "dove",
+        "new",
+        project_name,
+        "-d",
+        project_dialect,
+        "-r",
+        project_blockchain_api,
+        "-a",
+        project_address,
+    ];
+    let command_string: String = args.join(" ");
+    execute(args, base_folder.to_path_buf()).unwrap_or_else(|err| {
+        panic!(
+            "[COMMAND] {}\r\n[FOLDER] {}\r\n[ERROR] {}\r\n",
+            &command_string,
+            base_folder.display(),
+            err.to_string()
+        )
+    });
+    set_dependencies_local_move_stdlib(&project_folder);
+}
+// @dove build
+pub fn project_build(project_folder: &Path) {
+    let args = &["dove", "build"];
+    let command_string: String = args.join(" ");
+    execute(args, project_folder.to_path_buf()).unwrap_or_else(|err| {
+        panic!(
+            "[COMMAND] {}\r\n[FOLDER] {}\r\n[ERROR] {}\r\n",
+            &command_string,
+            project_folder.display(),
+            err.to_string()
+        )
+    });
+}
+pub fn project_remove(project_folder: &Path) {
+    if project_folder.exists() {
+        remove_dir_all(project_folder).unwrap_or_else(|err| {
+            panic!(
+                "[ERROR] Couldn't delete project directory: {}\r\n[MESSAGE] {}\r\n",
+                project_folder.display(),
+                err.to_string()
+            )
+        });
+    }
+}
+pub fn set_dependencies_local_move_stdlib(project_path: &Path) {
+    let move_stdlib = Path::new(".")
+        .canonicalize()
+        .unwrap()
+        .join("resources/test_move_project");
+    let mut dove_toml_path = project_path.to_path_buf();
+    dove_toml_path.push("Dove.toml");
+    let mut toml_value = read_to_string(&dove_toml_path)
+        .unwrap()
+        .parse::<Value>()
+        .unwrap();
+    {
+        let v = toml_value
+            .get_mut("package")
+            .unwrap()
+            .get_mut("dependencies")
+            .unwrap()
+            .as_array_mut()
+            .unwrap();
+        v.clear();
+        let mut dd = toml::map::Map::new();
+        dd.insert(
+            "path".to_string(),
+            Value::String(move_stdlib.to_str().unwrap().to_string()),
+        );
+        v.push(Value::Table(dd));
+    }
+    write_all(
+        &dove_toml_path,
+        toml::to_string(&toml_value).unwrap().as_str(),
+    )
+    .unwrap();
+}
+
+pub fn execute_dove_at(project_folder: &Path, args: &[&str]) {
+    execute(args, project_folder.to_path_buf()).unwrap_or_else(|err| {
+        panic!(
+            "[COMMAND] {}\r\n[FOLDER] {}\r\n[ERROR] {}\r\n",
+            &args.join(" "),
+            &project_folder.display(),
+            err.to_string()
+        )
+    });
+}
+pub fn execute_dove_at_wait_fail(project_folder: &Path, args: &[&str]) {
```

Comment:
nit: It's not clear why this function is called `_wait_fail` - it is not waiting for anything.

You can implement `execute_dove_at(....) -> Result<(), String>` and then use it in tests like `assert!(execute_dove_at(....).is_ok())` and `assert!(execute_dove_at(....).is_err())`. It's just an option though